### DISCUSSION TITLE
Question: How can we get Filter Token which appears in URL after application of PNP Search Filter?

### DISCUSSION THREAD

**Discussion by vitaj21** [microsoft-search/pnp-modern-search#3273] [General]
We are trying to create dynamically a URL for page which uses PNP Search Filter Webpart (Results webpart is also present of course). 

 
https://ms365thinking.blogspot.com/2023/08/pnp-modern-search-set-refiners-in-url.html= post from @kasperbolarsen suggests to use SP Search Query tool. However we were unsuccesfull at that. Do you know how to get these filter tokens dynamically, please?

I know that this was touched on in https://github.com/microsoft-search/pnp-modern-search/discussions/3209 but the question, of why this token is present now, remains unanswered.

**Comment by kasperbolarsen** [comment]
Can you share more info, like which kind of properties you are trying to use in the URL ?

**Comment by vitaj21** [comment]
Thank you for such a quick response!

I have a Search Results WebPart which is directed to a specific List called
Employee. This list has a Site Column which is mapped to RefinableString146.
Filtering via PNP Search Filters, using RefinableString146 as filter field,
works fine.
Resulting URL is for example:

tenant.sharepoint.com/sites/OrgChart/SitePages/OrgChart1.aspx?f=%5B%7B
"filterName"%3A"RefinableString146"%2C"values"%3A%5B%7B"name"%3A"HO%20CEO"%2C"value"%3A"%5C"
*ǂǂ484f2043454f*%5C""%2C"operator"%3A0%7D%5D%2C"operator"%3A"or"%7D%5D#

or
tenant.sharepoint.com/sites/OrgChart/SitePages/OrgChart1.aspx?f=%5B%7B
"filterName"%3A"RefinableString146"%2C"values"%3A%5B%7B"name"%3A"HO+Finance"%2C"value"%3A"%5C"
*ǂǂ484f2046696e616e6365*
%5C""%2C"operator"%3A0%7D%5D%2C"operator"%3A"or"%7D%5D

I want to generate these URLs on my own, as I want a long list of links
which will go to this page utilizing different values in the filter. As
these values are likely to change, I want to generate these links
dynamically.

How do I get the token e.g. ǂǂ484f2043454f ?


st 27. 9. 2023 v 18:50 odesílatel Kasper Larsen ***@***.***>
napsal:

**Comment by kasperbolarsen** [comment]
I have updated the screen shot of the SP Query Tool in the blog post as I have to agree that it isn't obvious how to get those values ( I couldn't remember how myself) 
But you have to enter the Managed Property in the "Refiners" box, run the query, and you should now be able to see the Refinement Tokens on the "Refinement Results" tab :-)

**Reply by vitaj21** [reply]
Thank you very much. We were successful in getting these tokens dynamically thanks to the SP Query Tool.
